File: CucumberJVM/Chapter15/15.7-StubDependencyPattern/After/src/test/java/com/wimp/specs/drivers/OrderingApiDriver.java
package com.wimp.specs.drivers;

import com.wimp.app.restapi.OrderResponse;
import com.wimp.app.restapi.PlaceOrderRequest;
import com.wimp.specs.support.LambdaAction;
import com.wimp.specs.support.RestApiContext;
import com.wimp.specs.support.TestAction;
import org.springframework.http.HttpMethod;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Component;

@Component
public class OrderingApiDriver {
    private final RestApiContext restApiContext;

    public OrderingApiDriver(RestApiContext restApiContext) {
        this.restApiContext = restApiContext;
    }

    public TestAction<OrderResponse> placeOrder(PlaceOrderRequest placeOrderRequest) {
        return new LambdaAction<>(() -> restApiContext.processRequest(
            "Place order",
            HttpMethod.POST,
            "/api/orders",
            placeOrderRequest,
            HttpStatus.CREATED,
            OrderResponse.class));
    }
}

File: Reqnroll/Chapter19/19.3-TestLogPattern/Before/WIMP.Specs/Drivers/OrderingApiDriver.cs
using System.Net;

using WIMP.App.Models;
using WIMP.App.RestApi;
using WIMP.Specs.Support;

namespace WIMP.Specs.Drivers;

public class OrderingApiDriver(RestApiContext restApiContext)
{
    public TestAction<Order> PlaceOrder(PlaceOrderRequest placeOrderRequest) =>
        new LambdaAction<Order>("Place order", placeOrderRequest, async () =>
            await restApiContext.ProcessRequest<Order>(
                "Place order", HttpMethod.Post, "/api/orders",
                placeOrderRequest, HttpStatusCode.Created));
}
